### PR TITLE
Ra-NeuS

### PR DESCRIPTION
**Pull request by wangyida**
Adding a new up_est_method ```z-axis```, so that the normalized poses could be always with an averaged up-direction of [0,0,1] as mentioned by [nerf2mesh](https://github.com/ashawkey/nerf2mesh/blob/main/nerf/colmap_provider.py#L30-L54), even when ground is missing in the training data. Such ```z-axis``` behaves similarly with ```ground``` when ground is included in the training data.

### `z-axis` up-direction
- [x] `z-axis` test for - ✅ mipnerf360 ✅ synthetic nerf
- [x] mix & match `z-axis` with - ✅ camera ✅ point ✅ lookat
- [x] test for pose match with [nerf2mesh](https://github.com/ashawkey/nerf2mesh)

### Extensions
- [x] SDF-to-radiance projection proposed by HF-NeuS and NSR
- [x] weight/alpha regularization 
- [x] add a configurable scale factor for camera translation vectors
- [x] directly load resized images (*2, *4, *8) when they are available
- [x] appearance embedding

### CODE REVIEW

**Comment by bennyguo** [comment]
Hi! It seems that this new up_est_method rotates the average camera up direction to +z. May I ask how this is different from `up_est_method == 'camera'`?

**Comment by wangyida** [comment]
> 

Hi Benny, the difference is that when camera is not positioned evenly surrouding the central object, _e.g._ the distance is not the same, ```camera``` setup will make the object tilted against z axis. the new ```axis-z``` setup will still position the object straight up along z axis.

File: datasets/colmap.py [hunk 1]
```diff
@@ -189,7 +248,7 @@ def setup(self, config, split):
                     img = Image.open(img_path)
                     img = img.resize(img_wh, Image.BICUBIC)
                     img = TF.to_tensor(img).permute(1, 2, 0)[...,:3]
-                    img = img.to(self.rank) if self.config.load_data_on_gpu else img.cpu()
```
Comment (bennyguo) [inline datasets/colmap.py:192]: I think you may forget to pull the latest version for this one :)

**Comment by bennyguo** [comment]
I see. Could you keep the config files unchanged and only modify the dataset? Also, is there an example scene illustrating the difference between the two methods so that I can test (for example some scene in Mip-NeRF360 dataset)?

File: datasets/colmap.py (hunk 1 above)
Comment (wangyida) [reply datasets/colmap.py:192]: ouch

**Comment by wangyida** [comment]
> 

Sure I can do it on bonsai scene

**Comment by wangyida** [comment]
pose compare
When ground is included, the new up direction option ```axis-z``` behaves similarly with ```ground```

**Comment by bennyguo** [comment]
Sorry I totally forgot about this PR 😂 Thank you very much for the newly added features, and I'll look into the code soon.

**Comment by wangyida** [comment]
Guess a [naive implementation](https://github.com/wangyida/instant-nsr-pl/blob/main/datasets/colmap.py#L58) for `point` center should be conditioned on points of sparse MVS reconstruction instead of cameras @bennyguo

**Comment by wangyida** [comment]
@bennyguo there are some new functions added, you mind review a bit further to see which of these modules are useful?

**Pull request closed without merging**